torch/csrc/jit/graph_executor.cpp

```diff
@@ -608,6 +608,13 @@ struct GraphExecutorImpl {
       for (Node* dnode : diff_nodes) {
         auto diff_graph = std::move(dnode->g(attr::Subgraph));
         Gradient gradient = differentiate(diff_graph);
+
+        // Re-run ConstantPropagation and Optimization with gradient.f
+        // In particular, we want to re-specialize symbolic_script
+        // Optionals relating to _grad_sum_to_size for the graph_fuser
+        ConstantPropagation(gradient.f);
```

Comment:
Any reason to not run propagate input shapes?

Reply:
?

Reply:
For some reason I have not fully understood yet, it seems to inhibit the graph fuser from fusing e.g. the mul(mask, *), div(, 1-prob). I didn't have a chance to investigate yesterday.
As this is closely related to #19988, I'll have to rebase anyway and update this.